tests/unit/provider/akamai/test_serviceController.py

```diff
@@ -0,0 +1,111 @@
+from unittest import TestCase
+import ddt
+import mock
+import uuid
+
+from nose.tools import assert_true
+
+from poppy.provider.akamai import services
+from tests.unit import base
+from poppy.model.helpers import domain
+from poppy.model.helpers import origin
+from poppy.model.service import Service
+from poppy.provider.akamai.services import ServiceController as akamai_service
+from poppy.model import service
+
+
+# from poppy.provider.akamai import driver
+
+@ddt.ddt
+class TestServiceController(base.TestCase):
+    def setUp(self):
+
+        print("1")
+        super(TestServiceController, self).setUp()
+        driver_patcher = mock.patch('poppy.provider.akamai.driver')
+        mock_driver = driver_patcher.start()
+        self.addCleanup(driver_patcher.stop)
+        print("2")
+        self.driver = mock_driver()
+        self.policy_client = self.driver.policy_api_client
+        self.ccu_client = self.driver.ccu_api_client
+        print("3")
+        self.driver.provider_name = 'Akamai'
+        self.driver.http_conf_number = 1
+        self.driver.akamai_https_access_url_suffix = str(uuid.uuid1())
+        self.driver.akamai_http_access_url_suffix = str(uuid.uuid1())
+        self.san_cert_cnames = [str(x) for x in range(7)]
+        self.driver.san_cert_cnames = self.san_cert_cnames
+        self.driver.metrics_resolution = 86400
+        self.controller = services.ServiceController(self.driver)
+        service_id = str(uuid.uuid4())
+        domains_old = domain.Domain(domain='cdn.poppy.org')
+        current_origin = origin.Origin(origin='poppy.org')
+        self.service_obj = Service(service_id=service_id,
+                                   name='poppy cdn service',
+                                   domains=[domains_old],
+                                   origins=[current_origin],
+                                   flavor_id='cdn')
+
+    @ddt.file_data('data_update_service.json')
+    def test__get_provider_access_url_shared_custom(self, service_json):
```

Comment:
I get why you put an additional leading underscore here -- but it should probably just be `test_get_provider_access_url_shared_custom`